### PR TITLE
Replace short-range LDA by libXC and cleanup

### PR DESCRIPTION
**Pull request by vanderhe**
Follow-up PR to #21:

- [x] reduce slateratom/sktwocnt input to subset that is actually needed
- [x] replace modified Burke short-range LDA by libXC and remove old subroutines
- [x] remove support for all xc-functionals which rely on ancient routines
- [x] re-indexing of remaining functionals
- [x] update [README](https://github.com/dftbplus/skprogs/blob/main/README.rst) to account for new libXC requirements (i.e. f03 interface)
- [x] update existing [skdef.hsd](https://github.com/dftbplus/skprogs/blob/main/examples/mio/skdef.hsd) example to new XCFunctional HSD block
- [x] add examples for long-range corrected functionals

Not tackled by this PR, yet important:
:negative_squared_cross_mark: replace ancient Broyden mixer by a more polished version
:negative_squared_cross_mark: replace mighty [EWEVGE](https://github.com/dftbplus/skprogs/blob/1a0babbc14f9e54362d8c57a9473eba0013d10be/slateratom/lib/diagonalizations.f90#L215) by something else (LAPACK linked anyway)
:negative_squared_cross_mark: replace f77 Lebedev–Laikov grids by a more modern f90 version

### CODE REVIEW

File: sktwocnt/lib/twocnt.f90 [hunk 1]
```diff
@@ -463,76 +463,45 @@ subroutine getskintegrals(t_integ, nRad, nAng, atom1, atom2, grid1, grid2, dots,
...
-        deallocate(densval1p, densval1pp, densval2p, densval2pp)
-      ! libXC treatment requires sigma = (nabla n).(nabla n) for GGA/LC
-      case(5:8)
-        allocate(densval1p(nGrid))
-        allocate(densval2p(nGrid))
+      ! care about correct 4pi normalization of density
+      rhor = getLibxcRho(densval)
+      allocate(vx(nGrid))
+      allocate(vc(nGrid))
+      if (iXC /= 1) then
```
Comment (aradi) [inline sktwocnt/lib/twocnt.f90:477]: use the type-based enum value also here.

File: sktwocnt/lib/twocnt.f90 [hunk 2]
```diff
@@ -217,53 +213,58 @@ subroutine get_twocenter_integrals(inp, imap, skham, skover)
     !! number of radial and angular integration abscissas
     integer :: nRad, nAng
 
-    if (inp%iXC == 5) then
+    if (inp%iXC == 1) then
```
Comment (aradi) [inline sktwocnt/lib/twocnt.f90:216]: If you changed this block any way, could you turn it please into a type based enumeration?
```
type :: TXcFunctionalsEnum
  integer :: LDA_PW91 = 1
  ...
end type TXcFunctionalsEnum

type(TXcFunctionalsEnum), parameter :: xcFunctionals = TXcFunctionalsEnum()

if (inp%iXC == xcFunctionals%LDA_PW91) then
```

File: sktwocnt/prog/input.f90 [hunk 3]
```diff
@@ -80,41 +77,35 @@ subroutine readInput(inp, fname)
     end select
 
     select case (iXC)
```
Comment (aradi) [inline sktwocnt/prog/input.f90:79]: type based enumeration instead of naked integers....

**Review by aradi — CHANGES_REQUESTED** [review]
I am fine with it, but please implement an enumeration for the xc-types on our side.

File: sktwocnt/lib/twocnt.f90 (hunk 1 above)
Comment (vanderhe) [reply sktwocnt/lib/twocnt.f90]: Good point :)

**Pull request merged**